### PR TITLE
Restore previous ServiceMonitor and Grafana dashboard ConfigMap templates and values

### PR DESCRIPTION
**Pull request by afalhambra-hivemq**
https://hivemq.kanbanize.com/ctrl_board/22/cards/22656/details/

### CODE REVIEW

File: charts/hivemq-swarm/templates/NOTES.txt [hunk 1]
```diff
@@ -22,10 +22,10 @@ To access the commander, please execute the following commands:
   echo "Commander is available via http://127.0.0.1:8080"
   kubectl --namespace {{ .Release.Namespace }} port-forward svc/{{ include "hivemq-swarm.fullname" . }}-api 8080:{{ .Values.service.port }}
 {{- end }}
-{{- if hasKey .Values "monitoring" }}
+{{- if hasKey .Values.monitoring "dedicated" }}
 
 *** Warning ***
-All `monitoring` values are deprecated and removed completely from the chart.
+Value `monitoring.dedicated` is deprecated and removed completely from the chart.
```
Comment (mhofsche) [inline charts/hivemq-swarm/templates/NOTES.txt:28]:
```suggestion
Value `monitoring.dedicated` is removed from the chart.
```

File: charts/hivemq-swarm/templates/NOTES.txt [hunk 2]
```diff
@@ -22,10 +22,10 @@ To access the commander, please execute the following commands:
...
 Please, consider creating your own Prometheus Monitoring stack manually. Check https://docs.hivemq.com/hivemq-swarm/latest/clustering.html#monitor for more help and guidance.
```
Comment (mhofsche) [inline charts/hivemq-swarm/templates/NOTES.txt:29]:
```suggestion
Please, create and setup your Prometheus Monitoring stack separately: check https://docs.hivemq.com/hivemq-swarm/latest/clustering.html#monitor for help.
```

File: charts/hivemq-swarm/templates/_helpers.tpl [hunk 3]
```diff
@@ -61,11 +61,21 @@ Create the name of the service account to use
...
     {{- .Values.namespaceOverride -}}
   {{- else -}}
     {{- .Release.Namespace -}}
   {{- end -}}
-{{- end -}}
\ No newline at end of file
+{{- end -}}
+
+{{/*
+Validates the required Prometheus Monitoring CRDs are installed in the Kubernetes cluster.
```
Comment (mhofsche) [inline charts/hivemq-swarm/templates/_helpers.tpl:73]:
```suggestion
Validates that the required Prometheus Monitoring CRDs are installed in the Kubernetes cluster.
```

File: charts/hivemq-swarm/templates/_helpers.tpl [hunk 4]
```diff
@@ -61,11 +61,21 @@ Create the name of the service account to use
...
+Otherwise, the installation will fail.
```
Comment (mhofsche) [inline charts/hivemq-swarm/templates/_helpers.tpl:74]:
```suggestion
If the CRD is not present the service monitor cannot be installed and the installation fails.
```

File: charts/hivemq-swarm/templates/_helpers.tpl [hunk 5]
```diff
@@ -61,11 +61,21 @@ Create the name of the service account to use
...
+*/}}
+{{- define "hivemq-swarm.validate-prometheus-monitoring-stack-installed" -}}
+{{- $isCRDPresent := .Capabilities.APIVersions.Has "monitoring.coreos.com/v1" }}
+{{- if not $isCRDPresent }}
+    {{- fail (printf "There is no ServiceMonitor CustomResourceDefinition available in your Kubernetes cluster. Prometheus Monitoring CRDs are required before installing the ServiceMonitor resource.\nCheck out https://docs.hivemq.com/hivemq-swarm/latest/clustering.html#monitor for more help and guidance.") }}
```
Comment (mhofsche) [inline charts/hivemq-swarm/templates/_helpers.tpl:79]:
```suggestion
    {{- fail (printf "There is no Prometheus ServiceMonitor CustomResourceDefinition (CRD) available in your Kubernetes cluster. Prometheus Monitoring CRDs are required before installing the ServiceMonitor resource.\nCheck out https://docs.hivemq.com/hivemq-swarm/latest/clustering.html#monitor for more help and guidance.") }}
```

File: charts/hivemq-swarm/files/dashboard.json
Comment (mhofsche) [inline charts/hivemq-swarm/files/dashboard.json:1]: should we name this file grafana-dashboard.json?

File: charts/hivemq-swarm/templates/configmap_grafana_dashboard.yaml [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+  namespace: {{ .Values.monitoring.dashboardNamespace }}
+  {{- else }}
+  namespace: {{ template "hivemq-swarm.namespace" . }}
+  {{- end }}
+  labels:
+    grafana_dashboard: "1"
+    {{- include "hivemq-swarm.labels" . | nindent 4 }}
+data:
+  swarm.json: |
+    {{- .Files.Get "files/dashboard.json" | nindent 4 }}
```
Comment (mhofsche) [inline charts/hivemq-swarm/templates/configmap_grafana_dashboard.yaml:17]: If we rename the file than this should reflect it.

**Review by mhofsche — APPROVED** [review]

**Review by Donnerbart — APPROVED** [review]

**Pull request merged**